liferay-headless/liferay-portal
# LPD-36166 Relations added between Objects and SystemObjects fail OpenAPI YAML Validation

**What's changed?:**

- The generated OpenAPIs for system objects and custom object relationships now correctly include required fields such as the `schema` and `description` attributes in the path parameters and response sections, ensuring proper validation and compliance with OpenAPI standards.
- The logic now dynamically retrieves the data type from the system object’s basic endpoints (e.g., `getUserAccount` for User) to ensure consistency. This resolves the issue where _id_ fields, although `long` in the database, were marked by default as `string` in the OpenAPI specification. In case of future updates to the `RestBuilder` (including proper swagger annotations), this approach ensures automatic alignment with the parameter type.
- A test has been added to validate that the OpenAPI is generated correctly, ensuring all necessary fields, such as `description` and `schema`, are present in the dynamic endpoints.

**How to test it?:**

- The behavior can be tested by creating a custom object, establishing a relationship with a system object (e.g., User), and verifying the OpenAPI output to confirm that the `schema` and `description` fields are properly defined. The OpenAPI can also be validated with the [Swagger & OpenAPI Online Validator](https://apitools.dev/swagger-parser/online)
- The added test ensures this behavior is consistent and correct.

See the following **Jira Ticket**: [LPD-36166](https://liferay.atlassian.net/jira/software/c/projects/LPD/issues/LPD-36166)
